### DISCUSSION TITLE
Packaging volume optimization

### DISCUSSION THREAD

**Discussion by Ya-hui** [wxt-dev/wxt#1075] [Q&A]
I am using wxt framework to develop Chrome plugins. There are two content_script scripts corresponding to two different domain names. Both of these content_script scripts use vue element-plus or some third-party libraries. When packaged, the size is very large. I want to extract the third-party package and not package it into content_script, but load it through <script src=><-script>

**Comment by Ya-hui** [comment]
manifest.json

// This is currently
```json
{
    "matches": [
        "*://re.1688.com/*"
    ],
    "js": [
        "content-scripts/ali.js"
    ],
    "css": [
        "content-scripts/ali.css"
    ]
}

```

Want to achieve
```json
{
    "matches": [
        "*://re.1688.com/*"
    ],
    "js": [
        "vue.js",
        "element-plus.js"
        "content-scripts/ali.js"
    ],
    "css": [
        "style.css"
        "content-scripts/ali.css"
    ]
}

```

**Reply by aklinker1** [reply]
There isn't currently a way to split content scripts like this.

If you really need to support chunking and share your code, it is possible to force Vite into generating chunks: https://github.com/wxt-dev/examples/tree/main/examples/esm-content-script-ui

But I wouldn't recommend it. Bundle size isn't as important in web extensions as it is in regular web development since files are loaded from the disc rather than the network. If I were you, I'd leave it as is and wait for ESM content script support: #357

**Reply by Ya-hui** [reply]
So can CSS be split? For example, common CSS styles
```json
"css": [
        "style.css",
       "element-plus.css",
        "content-scripts/ali.css"
    ]
```

**Reply by aklinker1** [reply]
You could do this manually by transforming the manifest in the `build:manifestGenerated` hook, but there's no built-in way to do this.

https://wxt.dev/guide/essentials/config/hooks.html#adding-hooks
